Fix new warnings in CE2021

mosteo opened the pull request.

mosteo committed:
Initial fixes for CE2021

mosteo committed:
Resolve conflicts with CE2020

Fabien-Chouteau commented:
You also change the CI runs by using the new version of the toolchain GitHub Action: `uses: ada-actions/toolchain@ce2021`

mosteo committed:
Update CI scripts to use CE2021

mosteo committed:
Merge branch 'master' into fix/ce2021

mosteo commented:
> You also change the CI runs by using the new version of the toolchain GitHub Action: `uses: ada-actions/toolchain@ce2021`

Is it normal that it is failing for macOS?

Fabien-Chouteau commented:
> Is it normal that it is failing for macOS?

Yes, I forgot that there is not ce2021 for macOS so you have to use ce2020 there.

mosteo committed:
Revert CE version for macOS CI tests

mosteo committed:
Merge conflicts resolved via web interface

mosteo committed:
Remove conflicting Append_Vector

mosteo committed:
Remove unused entities in Windows

mosteo committed:
Fix bug uncovered by CE2021

mosteo committed:
Fix timestamps compare when mixing G.OS_Lib/A.Dirs

mosteo committed:
Update pins for self-build

mosteo committed:
Disable self-build, that uses CE2021

It cannot succeed until the patches in this very PR are merged, as it is
attempting to build the old code with the new compiler in the setup-alire
action.

mosteo committed:
More fixes for relative paths in softlinks

mosteo committed:
Remove unused entities in Windows build
